File: leetcode/golang/LCR_180_he_wei_sde_lian_xu_zheng_shu_xu_lie_lcof.go
func fileCombination(target int) [][]int {
    var res [][]int
    var tmp []int
    sum := 0
    for i, j := 1, 1; j <= target / 2 + 1; j++ {
        sum += j
        tmp = append(tmp, j)
        for sum > target {
            sum -= i
            i++
        }
        if sum == target {
            res = append(res, tmp[i-1:j])
        }
    }
    return res
}

File: leetcode/c++/LCR_180_he_wei_sde_lian_xu_zheng_shu_xu_lie_lcof.cpp
// https://leetcode.cn/problems/he-wei-sde-lian-xu-zheng-shu-xu-lie-lcof

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> fileCombination(int target) {
        vector<vector<int>> res;
        vector<int> tmp;
        int sum = 0;
        for (int i = 1, j = 1; j <= target / 2 + 1; j++) {
            sum += j;
            tmp.push_back(j);
            while (sum > target) {
                sum -= i++;
            }
            if (sum == target) {
                vector<int> v(tmp.begin()+i-1, tmp.begin()+j);
                res.push_back(v);
            }              
        }
        return res;
    }
};

// 想不到数学办法，直接滑动窗口，O(n),O(1)